### PR TITLE
feat: prompt modal on edit

### PR DESCRIPTION
**Pull request by AmarTrebinjac**
## Changes
Added prompt modal on edit, as well as moving remaining create/edit flows into usePostToSquad.

Originally when I updated this I tried to consolidate everything in as few callbacks as possible. That was a mistake. Made it confusing to work with and caused some weird behaviors, so I decided on just giving each action its own callback.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-584 #done
MI-619 #done

[MI-584]: https://dailydotdev.atlassian.net/browse/MI-584?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AmarTrebinjac** [comment]
Anyone know why preview isn't working properly?

File: packages/shared/src/graphql/posts.ts [hunk 1]
```diff
@@ -553,24 +553,28 @@ export interface EditPostProps {
...
+      sourceId: string;
+      type: PostType;
+      sharedPostId?: string;
+      externalLink?: string;
+      imageUrl?: string;
+      image?: File;
+      commentary?: string;
+      postId?: string;
+    }
+  | { postId: string; type?: PostType; image: File };
```
Comment (sshanzel) [inline packages/shared/src/graphql/posts.ts:577]: Why do we need only three props for edit?

Comment (AmarTrebinjac) [reply packages/shared/src/graphql/posts.ts:577]: @sshanzel Totally forgot about this one! You're right, we should have the rest of the props, too!

**Review by rebelchris — APPROVED** [review]
Besides Lee's feedback looks good

**Review by ilasw — APPROVED** [review]
Great! Looks good to me, tested in local env and works like a charm :) 

Just coordinate with @sshanzel before merge pls

**Pull request merged**